front/lib/api/assistant/generation.ts

```diff
@@ -92,15 +92,16 @@ export async function renderConversationForModelMultiActions({
 
       for (const action of actions) {
         const stepIndex = action.step;
-        steps[stepIndex] = steps[stepIndex] || emptyStep();
+        steps[stepIndex] = stepIndex in steps ? steps[stepIndex] : emptyStep();
```

Comment:
So this becomes the new way to check if an object has a key. Even though it's probably right, this requires to ensure that all engineers follow this pattern. Did we consider this? https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess

Reply:
We did consider it yesterday but we have 756 errors to fix if we enable this check, so we could do it but in a separate PR.

Reply:
But then what's the expectations, should everyone start using this "in" pattern?

Reply:
Once the linter is on, if you don't use it, you will get an error, so you will have to use it.